drivers/clock_management/nxp_syscon/nxp_syscon_div.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/clock_management/clock_driver.h>
+
+#define DT_DRV_COMPAT nxp_syscon_clock_div
+
+struct syscon_clock_div_config {
+	uint8_t mask_width;
+	const struct clk *parent;
+	volatile uint32_t *reg;
+};
+
+
+static int syscon_clock_div_get_rate(const struct clk *clk_hw)
+{
+	const struct syscon_clock_div_config *config = clk_hw->hw_data;
+	int parent_rate = clock_get_rate(config->parent);
+	uint8_t div_mask = GENMASK((config->mask_width - 1), 0);
+
+	if (parent_rate <= 0) {
+		return parent_rate;
+	}
+
+	/* Calculate divided clock */
+	return parent_rate / ((*config->reg & div_mask) + 1);
```

Comment:
I don't know how picky you need to be... but you may want to use `DIV_ROUND_CLOSEST` here

Reply:
I think it is ok as is- on platforms using this driver code size should matter more than rounding the clock correctly. Plus, I think we'd rather round down for any type of clock calculation.